packages/clay-drop-down/src/Menu.tsx

```diff
@@ -187,6 +193,10 @@ const ClayDropDownMenu = React.forwardRef<HTMLDivElement, IProps>(
 
 		useEffect(() => {
 			const handleClick = (event: MouseEvent) => {
+				if (!closeOnClickOutside) {
+					return;
+				}
+
```

Comment:
@bryceosterhaus maybe we can move this validation when attaching the event, so we can avoid adding the event that will not be used. What do you think?

Reply:
Good idea!